internal/codecs/datetime_test.go

```diff
@@ -0,0 +1,56 @@
+// This source file is part of the EdgeDB open source project.
+//
+// Copyright 2020-present EdgeDB Inc. and the EdgeDB authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codecs
+
+import (
+	"testing"
+	"unsafe"
+
+	"github.com/edgedb/edgedb-go/internal/buff"
+	types "github.com/edgedb/edgedb-go/internal/edgedbtypes"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLocalDateTimeDecode(t *testing.T) {
+	r := buff.SimpleReader([]byte{
```

Comment:
BTW, I'm not sure if we really need low-level byte IO tests like this. Up to you, but wouldn't it be enough to just have high-level functional tests where we query the DB with these types?

Reply:
Yes it would be enough to identify that something is wrong. I feel like tests like this can help identify what specifically is wrong. That said, I am open to getting rid of them. They do incure a certain amount of development over head.

Reply:
> They do incure a certain amount of development over head.

Yeah, this is why I'm asking. We don't have tests like this in edgedb-js or edgedb-python. So if maintaining them is tedious I'd drop them. Also looking at `x00, 0x02, 0x2b, 0x35, 0x9b, 0xc4, 0x10, 0x00` doesn't show me a clear picture of the data layout, I'd still need to debug it to understand what went wrong.